### DISCUSSION TITLE
Point Intersects GeoJson MultiPolyon with a hole

### DISCUSSION THREAD

**Discussion by emsmithmsft** [NetTopologySuite/NetTopologySuite#692] [Q&A] [answered]
I have a collection of GeoJson features with MultiPolygon coordinates and a list of points which I am determining which feature they belong to.
Each MultiPolygon is mutually exclusive, however some have a hole in the coordinates to carve out sections that are otherwise internal. These seem to not be honored by the NetTopologySuite Intesects method. If a point is within the hole, Intersects still returns true.

Various comments in the repo say "The polygons must conform to the assertions specified in the [OpenGIS Simple Features Specification for SQL](http://www.opengis.org/techno/specs.htm). I had found the section 6.1.5.1 in the doc mentions that polygons may contain holes - but it does not go into detail on formatting.

The [formatting for holes in GeoJson I am referencing is here](https://datatracker.ietf.org/doc/html/rfc7946#appendix-A.3). 
 NetTopologySuite has a module for serializing and deserializing GeoJson, so I assumed it was also supported for other functionality of the library.  

Is it correct that this scenario is not supported?
If so, do you have a recommendation for how I can best determine if a Point is within the MutliPolygon excluding the hole region? GeoJson.Net and GeoJson.Text which I have also looked at do not have intersect functionality.

**Comment by FObermaier** [comment, marked as answer]
This should not be the case, NTS itself does not care about ring orientation for its computation at all.
```C#
const string does = " intersects with ";
const string doesnot = " doesn't intersect with ";

var rdr = new NetTopologySuite.IO.WKTReader();
var pg = rdr.Read("MultiPolygon(((0 0, 10 0, 10 10, 0 10, 0 0), (2 2, 8 2, 8 8, 2 8, 2 2)), ((20 0, 30 0, 30 10, 20 10, 20 0), (22 2, 28 2, 28 8, 22 8, 22 2)))");
var p1 = rdr.Read("Point(1 1)");
var p2 = rdr.Read("Point(5 5)");

Console.WriteLine("Interior CCW");
Console.WriteLine($"{p1}{(p1.Intersects(pg) ? does : doesnot)}{pg}");
Console.WriteLine($"{p2}{(p2.Intersects(pg) ? does : doesnot)}{pg}");

pg = rdr.Read("MultiPolygon(((0 0, 10 0, 10 10, 0 10, 0 0), (2 2, 2 8, 8 8, 8 2, 2 2)), ((20 0, 30 0, 30 10, 20 10, 20 0), (22 2, 22 8, 28 8, 28 2, 22 2)))");
Console.WriteLine("Interior CW");
Console.WriteLine($"{p1}{(p1.Intersects(pg) ? does : doesnot)}{pg}");
Console.WriteLine($"{p2}{(p2.Intersects(pg) ? does : doesnot)}{pg}");
```
```
No692:
Interior CCW
POINT (1 1) intersects with MULTIPOLYGON (((0 0, 10 0, 10 10, 0 10, 0 0), (2 2, 8 2, 8 8, 2 8, 2 2)), ((20 0, 30 0, 30 10, 20 10, 20 0), (22 2, 28 2, 28 8, 22 8, 22 2)))
POINT (5 5) doesn't intersect with MULTIPOLYGON (((0 0, 10 0, 10 10, 0 10, 0 0), (2 2, 8 2, 8 8, 2 8, 2 2)), ((20 0, 30 0, 30 10, 20 10, 20 0), (22 2, 28 2, 28 8, 22 8, 22 2)))
Interior CW
POINT (1 1) intersects with MULTIPOLYGON (((0 0, 10 0, 10 10, 0 10, 0 0), (2 2, 2 8, 8 8, 8 2, 2 2)), ((20 0, 30 0, 30 10, 20 10, 20 0), (22 2, 22 8, 28 8, 28 2, 22 2)))
POINT (5 5) doesn't intersect with MULTIPOLYGON (((0 0, 10 0, 10 10, 0 10, 0 0), (2 2, 2 8, 8 8, 8 2, 2 2)), ((20 0, 30 0, 30 10, 20 10, 20 0), (22 2, 22 8, 28 8, 28 2, 22 2)))
```

Can you provide sample data/code to prove your issue?

**Comment by emsmithmsft** [comment]
You are correct. When trying to get a simplified example of my issue I realized that it was caused by incorrectly deserializing the Coordinates of the geojson into a multipolygon rather than an issue with Intersects. 
We had been deserializing into a custom class where the Coordinates were a List<List<List<List<double>>>> and translating them to a NTS MultiPoligon - and that was flattening out a layer so the 'hole' polygon was being treated as a separate polygon. 

When I deserialize straight into a NTS FeatureCollection it is working as expected, and I can still access the other properties on the geojson objects through Attributes. 

Thanks!

**Reply by FObermaier** [reply]
Using `NetTopologySuite.IO.GeoJSON` or  `NetTopologySuite.IO.GeoJSON4STJ` nuget package was not an option?

**Reply by emsmithmsft** [reply]
I assume that whoever wrote this before did not notice that nuget pacakge. I am using the NTS GeoJsonSerializer now 😊
